core/src/main/scala/besom/internal/BesomModule.scala

```diff
@@ -3,30 +3,63 @@ package besom.internal
+    * program but are defined outside the `Pulumi.run` block should have the following using clause: `(using Context)`
+    * or `(using besom.Context)` using a fully qualified name of the type.
+    *
+    * Example:
+    * {{{
+    * import besom.*
+    * import besom.api.aws
+    *
+    * @main def run = Pulumi.run {
+    *   for
+    *     bucket <- aws.s3.Bucket("my-bucket")
+    *   yield exports(
+    *     bucketUrl = bucket.websiteEndpoint
+    *   )
+    * }
+    * }}}
+    *
+    * @param program
+    *   the program to run
+    */
   def run(program: Context ?=> Output[Exports]): Unit =
     val everything: Result[Unit] = Result.scoped {
       for
-        _              <- BesomLogger.setupLogger()
-        runInfo        <- RunInfo.fromEnv
-        _              <- logger.debug(s"Besom starting up in ${if runInfo.dryRun then "dry run" else "live"} mode.")
-        taskTracker    <- TaskTracker()
-        monitor        <- Result.resource(Monitor(runInfo.monitorAddress))(_.close())
-        engine         <- Result.resource(Engine(runInfo.engineAddress))(_.close())
-        _              <- logger.debug(s"Established connections to monitor and engine, spawning streaming pulumi logger.")
-        logger         <- Result.resource(BesomLogger(engine, taskTracker))(_.close())
-        config         <- Config.forProject(runInfo.project)
+        _           <- BesomLogger.setupLogger()
+        runInfo     <- RunInfo.fromEnv
+        _           <- logger.debug(s"Besom starting up in ${if runInfo.dryRun then "dry run" else "live"} mode.")
+        taskTracker <- TaskTracker()
+        monitor     <- Result.resource(Monitor(runInfo.monitorAddress))(_.close())
+        engine      <- Result.resource(Engine(runInfo.engineAddress))(_.close())
+        _           <- logger.debug(s"Established connections to monitor and engine, spawning streaming pulumi logger.")
+        logger      <- Result.resource(BesomLogger(engine, taskTracker))(_.close())
+        config      <- Config.forProject(runInfo.project)
         featureSupport <- FeatureSupport(monitor)
-        _              <- logger.trace(s"Environment:\n${sys.env.toSeq.sortBy(_._1).map((k, v) => s"$k: $v").mkString("\n")}")
-        _              <- logger.debug(s"Resolved feature support, spawning context and executing user program.")
-        ctx            <- Context(runInfo, taskTracker, monitor, engine, logger, featureSupport, config)
-        userOutputs    <- program(using ctx).map(_.toResult).getValueOrElse(Result.pure(Struct()))
-        _              <- Stack.registerStackOutputs(runInfo, userOutputs)(using ctx)
-        _              <- ctx.waitForAllTasks
+        _   <- logger.trace(s"Environment:\n${sys.env.toSeq.sortBy(_._1).map((k, v) => s"$k: $v").mkString("\n")}")
+        _   <- logger.debug(s"Resolved feature support, spawning context and executing user program.")
+        ctx <- Context(runInfo, taskTracker, monitor, engine, logger, featureSupport, config)
```

Comment:
why are these 3 lines indented this way? could you set maxColumns to 140 and reformat? I know it's asinine of me to demand ascii graphics in code but I really want to avoid dumb changes in the single reformat commit.